gnucash/import-export/import-main-matcher.c

```diff
@@ -976,8 +976,8 @@ gnc_gen_trans_onButtonPressed_cb (GtkTreeView *treeview,
                 GList* selected;
                 GtkTreeModel *model;
                 selected = gtk_tree_selection_get_selected_rows (selection, &model);
-                get_action_for_path (selected->data, model);
-                if (get_action_for_path (selected->data, model) == GNCImport_ADD)
+                if (gtk_tree_path_get_depth ((GtkTreePath *)selected->data) == 1 &&
+                    get_action_for_path (selected->data, model) == GNCImport_ADD)
```

Comment:
You've added another condition here as well as removed the extra call to `get_action_for_path`. The commit message should say why.